### PR TITLE
zdb long options

### PR DESCRIPTION
**Pull request by richardelling**
Add long options to zdb command.
Includes updates to ZTS tests and man page.
Includes fix for known failures of `zdb_006_pos`

### Motivation and Context

Resolves https://github.com/zfsonlinux/zfs/issues/8739

### Description

Adds long option names for existing zdb options.
Adds `--help`
Includes support for future long options that will be needed for the
draid project.

### How Has This Been Tested?

ZTS tests included and pass

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: cmd/zdb/zdb.c [hunk 1]
```diff
@@ -6042,12 +6118,27 @@ main(int argc, char **argv)
 	argc -= optind;
 	argv += optind;
 
-	if (argc < 2 && dump_opt['R'])
+	if (getenv("ZFS_OPTIONS_DEBUG") != NULL) {
```
Comment (behlendorf) [inline cmd/zdb/zdb.c]: Can you either document `ZFS_OPTIONS_DEBUG` in the `zdb` man page, or if it's intended to only be used for internal ZTS testing add a comment to that effect here.

File: cmd/zdb/zdb.c [hunk 2]
```diff
@@ -139,88 +147,97 @@ usage(void)
...
 	    "\t%s -m [-AFLPX] [-e [-V] [-p <path> ...]] [-t <txg>] "
-	    "[-U <cache>]\n\t\t<poolname> [<vdev> [<metaslab> ...]]\n"
+	    "[-U <cachefile>]\n\t\t<poolname> [<vdev> [<metaslab> ...]]\n"
 	    "\t%s -O <dataset> <path>\n"
-	    "\t%s -R [-A] [-e [-V] [-p <path> ...]] [-U <cache>]\n"
+	    "\t%s -R [-A] [-e [-V] [-p <path> ...]] [-U <cachefile>]\n"
 	    "\t\t<poolname> <vdev>:<offset>:<size>[:<flags>]\n"
 	    "\t%s -E [-A] word0:word1:...:word15\n"
-	    "\t%s -S [-AP] [-e [-V] [-p <path> ...]] [-U <cache>] "
+	    "\t%s -S [-AP] [-e [-V] [-p <path> ...]] [-U <cachefile>] "
```
[the remark this answers is not in the document]
Comment (richardelling) [reply cmd/zdb/zdb.c]: good idea, I'll add a test case for this

File: cmd/zdb/zdb.c (hunk 1 above)
Comment (richardelling) [reply cmd/zdb/zdb.c]: it is for testing only, I'll add references to make it clearer

**Comment by behlendorf** [comment]
@richardelling how would you like to move forward with this.  If you could address the handful of review comments and get it rebased we can get it merged.

**Comment by richardelling** [comment]
I've got too many deadlines this week, was planning to rebase and address the other comments this weekend.

**Comment by behlendorf** [comment]
@richardelling it'd be great if you could dust this off, rebase, and address the comments.

**Comment by ahrens** [comment]
@richardelling When do you think you'll have time to get back to this?

**Comment by richardelling** [comment]
after supercomputing conference

**Comment by behlendorf** [comment]
Closing.  @richardelling if you get a chance to get back to this please feel free to open a new PR.

**Pull request closed without merging**